docs/develop/auth.md

```diff
@@ -77,6 +76,27 @@ func AuthHandler(ctx context.Context, token string) (auth.UID, error) {
 }
 ```
 
+
+## Returning custom error messages on auth handlers
+
+When you want to return a custom error message in the auth handler e.g To return an `unauthenticated` error message:
```

Comment:
In the same vein as above, I think we should set the stage for the reader and give a bit more detail:

```suggestion
When a token doesn't match your auth rules (for example if it's expired, the token has been revoked, or the token is invalid), you should return a non-nil error from the auth handler.

Encore passes the error message on to the user when you use [Encore's built-in error package](errors), so we recommend using that with the error code `Unauthenticated` to communicate what happened. For example: 
```